python/sparkdl/graph/input.py

```diff
@@ -0,0 +1,317 @@
+    Notice that one should either provide the `signature_def_key` or provide both
+    `feed_names` and `fetch_names`. Please set the unprovided values to None.
+
+    :param signature_def_key: str, name of the mapping contained inside the `signature_def`
+                              from which we retrieve the signature key to tensor names mapping.
+    :param feed_names: list, names of the input tensors.
+    :param fetch_names: list, names of the output tensors.
+    """
+    assert (feed_names is None) == (fetch_names is None), \
+        'feed_names and fetch_names, if provided must appear together'
+    assert (feed_names is None) != (signature_def_key is None), \
+        'must either provide feed_names or singnature_def_key'
+
+    graph = tf.Graph()
+    with tf.Session(graph=graph) as sess:
+        tag_sets = tag_set.split(',')
+        meta_graph_def = tf.saved_model.loader.load(sess, tag_sets, saved_model_dir)
+
+        if signature_def_key is not None:
+            sig_def = tf.contrib.saved_model.get_signature_def_by_key(meta_graph_def,
+                                                                      signature_def_key)
+            gin = _build_with_sig_def(sess=sess, graph=graph, sig_def=sig_def)
+        else:
+            gin = _build_with_feeds_fetches(sess=sess, graph=graph, feed_names=feed_names,
+                                            fetch_names=fetch_names)
+    return gin
+
+
+def _build_with_sig_def(sess, graph, sig_def):
+    # pylint: disable=protected-access,attribute-defined-outside-init
+    assert sig_def, \
+        'signature_def {} provided, '.format(sig_def) + \
+        'but failed to find it from the meta_graph_def'
+
+    with sess.as_default(), graph.as_default():
+        feed_mapping = {}
+        feed_names = []
+        for sigdef_key, tnsr_info in sig_def.inputs.items():
+            tnsr_name = tnsr_info.name
+            feed_mapping[sigdef_key] = tnsr_name
+            feed_names.append(tnsr_name)
+
+        fetch_mapping = {}
+        fetch_names = []
+        for sigdef_key, tnsr_info in sig_def.outputs.items():
+            tnsr_name = tnsr_info.name
+            fetch_mapping[sigdef_key] = tnsr_name
+            fetch_names.append(tnsr_name)
+
+        for tnsr_name in feed_names:
+            assert tfx.get_op(graph, tnsr_name), \
+                'requested tensor {} but found none in graph {}'.format(tnsr_name, graph)
+        fetches = [tfx.get_tensor(graph, tnsr_name) for tnsr_name in fetch_names]
+        graph_def = tfx.strip_and_freeze_until(fetches, graph, sess)
+
+    return TFInputGraph(graph_def=graph_def, input_tensor_name_from_signature=feed_mapping,
+                        output_tensor_name_from_signature=fetch_mapping)
+
+
+def _build_with_feeds_fetches(sess, graph, feed_names, fetch_names):
+    # pylint: disable=protected-access,attribute-defined-outside-init
+    assert (feed_names is not None) and (fetch_names is not None), \
```

Comment:
I don't think this check is necessary, but if you must, I'd do it one by one so it's more obvious when the error is thrown:
```
assert (feed_names is not None), "must provide feed_names"
assert (fetch_names is not None), "must provide fetch names"
```
also, is it bad if they are empty? if so, should check for that and modify the error msg above to say "must provide non-empty ..."